src/dodal/devices/i04/transfocator.py

```diff
@@ -18,65 +22,80 @@ class Transfocator(Device):
+
+        # The value hasn't changed so assume the device is already set up correctly
+        async for value in observe_value(self.predicted_vertical_num_lenses):
+            await set_based_on_predicition(value)
+            if have_we_done_it:
+                break
+
+    async def polling_wait_on_start_rbv(self, for_value):
         # For some reason couldn't get monitors working on START_RBV
         # (See https://github.com/DiamondLightSource/dodal/issues/152)
         start_time = time()
         while time() < start_time + self.TIMEOUT:
-            RBV_value = self.start_rbv.get()
+            RBV_value = await self.start_rbv.get_value()
             if RBV_value == for_value:
                 return
-            sleep(self._POLLING_WAIT)
+            await asyncio.sleep(self._POLLING_WAIT)
 
         # last try
-        if self.start_rbv.get() != for_value:
+        if await self.start_rbv.get_value() != for_value:
             raise TimeoutError()
 
-    def set(self, beamsize_microns: float) -> DeviceStatus:
+    @AsyncStatus.wrap
+    async def set(self, beamsize_microns: float) -> None:
         """To set the beamsize on the transfocator we must:
         1. Set the beamsize in the calculator part of the transfocator
         2. Get the predicted number of lenses needed from this calculator
         3. Enter this back into the device
         4. Start the device moving
         5. Wait for the start_rbv goes high and low again
         """
-        subscriber: int
-        status = DeviceStatus(self, timeout=self.TIMEOUT)
-
-        def set_based_on_predicition(old_value, value, *args, **kwargs):
-            if not math.isclose(old_value, value, abs_tol=1e-8):
-                self.predicted_vertical_num_lenses.unsubscribe(subscriber)
-
-                # We can only put an integer number of lenses in the beam but the
-                # calculation in the IOC returns the theoretical float number of lenses
-                value = round(value)
-                LOGGER.info(f"Transfocator setting {value} filters")
-                self.number_filters_sp.set(value).wait()
-                self.start.set(1).wait()
-                self.polling_wait_on_start_rbv(1)
-                self.polling_wait_on_start_rbv(0)
-            # The value hasn't changed so assume the device is already set up correctly
-            status.set_finished()
+        self.latest_pred_vertical_num_lenses = (
+            await self.predicted_vertical_num_lenses.get_value()
+        )
 
         LOGGER.info(f"Transfocator setting {beamsize_microns} beamsize")
-        if self.beamsize_set_microns.get() != beamsize_microns:
-            subscriber = self.predicted_vertical_num_lenses.subscribe(
-                set_based_on_predicition, run=False
+
+        if await self.beamsize_set_microns.get_value() != beamsize_microns:
+            await asyncio.gather(
+                self.beamsize_set_microns.set(beamsize_microns),
```

Comment:
Should: Add a comment here to explain that changing that at the IOC level, `beamsize_set_microns` will cause `predicted_vertical_num_lenses` to change

Reply:
Added a comment to describe what happens at the IOC level 👍